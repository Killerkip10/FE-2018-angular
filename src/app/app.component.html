<app-header></app-header>

<mat-drawer-container class="container">
  <mat-drawer *ngIf="tokenService.getToken()" mode="side" opened>
    <mat-nav-list>
      <a mat-list-item [routerLink]="['user-profile']" routerLinkActive="active">Profile</a>
      <a mat-list-item [routerLink]="['user-list']" routerLinkActive="active">Users list</a>

    </mat-nav-list>
  </mat-drawer>

  <mat-drawer-content>
    <router-outlet></router-outlet>
  </mat-drawer-content>
</mat-drawer-container>

<app-footer></app-footer>
