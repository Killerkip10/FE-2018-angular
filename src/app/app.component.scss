app-header{
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 2;
}
.container {
  margin-top: 56px;
  min-height: calc(100vh - 112px);
  mat-drawer{
    position: fixed;
    top: 56px;
    bottom: inherit;
  }
  .active {
    background-color: rgba(0,0,0,.04);
  }
}
