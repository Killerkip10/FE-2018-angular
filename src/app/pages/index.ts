export * from './user-profile/user-profile-page.module';
export * from './login/login-page.module';
export * from './forgot-password/forgot-password-page.module';
export * from './user-list/user-list.module';
