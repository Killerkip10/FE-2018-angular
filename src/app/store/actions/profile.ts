import {Action} from '@ngrx/store';

import {User} from '../../models';

export const GET_PROFILE = '[Profile] Get profile';
export const EDIT = '[Profile] Update';
export const DELETE = '[Profile] Delete';
export const SUCCESS = '[Profile] Success';
export const ERROR = '[Profile] Error';

export class GetProfile implements Action {
  public readonly type = GET_PROFILE;
}
export class EditProfile implements Action {
  public readonly type = EDIT;

  constructor(public profile: User) {}
}
export class DeleteProfile implements Action {
  public readonly type = DELETE;
}
export class Success implements Action {
  public readonly type = SUCCESS;

  constructor(public profile: User) {}
}
export class Error implements Action {
  public readonly type = ERROR;

  constructor(public err: string) {}
}

export type Action = GetProfile
  | EditProfile
  | DeleteProfile
  | Success
  | Error;
