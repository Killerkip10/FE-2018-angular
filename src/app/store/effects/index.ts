import {ProfileEffects} from './profile';
import {UsersEffects} from './users';

export const effects = [
  ProfileEffects,
  UsersEffects
];
