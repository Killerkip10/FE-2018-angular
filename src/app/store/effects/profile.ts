import {Injectable} from '@angular/core';
import {HttpClient, HttpHeaders, HttpResponse} from '@angular/common/http';
import {Actions, Effect, ofType} from '@ngrx/effects';
import {map, switchMap} from 'rxjs/operators';

import {User} from '../../../../server/models';
import {urlConfig} from '../../config';
import {
  GET_PROFILE,
  EDIT,
  Edit,
  Success
} from '../actions/profile';

const options = {
  headers: new HttpHeaders({'content-type': 'application/json'}),
  observe: 'response',
  withCredentials: true,
  params: null
};

@Injectable()
export class ProfileEffects {
  constructor(
    private actions$: Actions,
    private http: HttpClient
  ) {}

  @Effect()
  loadProfile$ = this.actions$
    .pipe(
      ofType(GET_PROFILE),
      switchMap(() => this.http.get<HttpResponse<User>>(urlConfig.getProfile, options as object)),
      map(resp => new Success(resp.body))
    );

  @Effect()
  editProfile$ = this.actions$
    .pipe(
      ofType(EDIT),
      switchMap((action: Edit) => this.http.put<HttpResponse<User>>(urlConfig.editProfile, action.profile, options as object)),
      map(resp => new Success(resp.body))
    );
}
