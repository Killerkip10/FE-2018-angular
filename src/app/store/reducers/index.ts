import {ActionReducerMap} from '@ngrx/store';

import * as fromUsers from './users';
import * as fromProfile from './profile';

export interface State {
  users: fromUsers.State;
  profile: fromProfile.State;
}

export const reducers: ActionReducerMap<State> = {
  users: fromUsers.reducer,
  profile: fromProfile.reducer
};
