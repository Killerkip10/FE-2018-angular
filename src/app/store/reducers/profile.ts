import {Map} from 'immutable';

import * as profileAction from '../actions/profile';
import {createFeatureSelector, createSelector} from '@ngrx/store';

import {User} from '../../models';

export interface State {
  profile: User;
  loaded: boolean;
  err: string;
}

export type MapState = boolean | string | object;

export const initialState = Map({
  profile: {},
  loaded: false,
  err: ''
});

export function reducer(state = initialState, action: profileAction.Action) {
  switch (action.type) {
    case profileAction.GET_PROFILE: {
      return state.set('loaded', false);
    }
    case profileAction.DELETE: {
      return state
        .set('profile', {})
        .set('loaded', false);
    }
    case profileAction.SUCCESS: {
      return state
        .set('profile', action.profile)
        .set('loaded', true)
        .set('err', '');
    }
    case profileAction.ERROR: {
      return state.set('err', action.err);
    }
    default: {
      return state;
    }
  }
}

export const getProfileMapState = createFeatureSelector<Map<string, MapState>>('profile');
export const getProfileJsState = createSelector(getProfileMapState, state => state.toJS());
