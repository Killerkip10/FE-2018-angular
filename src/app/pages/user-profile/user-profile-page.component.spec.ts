import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {CommonModule} from '@angular/common';
import {HttpClientModule} from '@angular/common/http';
import {TranslateModule} from '@ngx-translate/core';
import {of} from 'rxjs';

import {StoreModule} from '@ngrx/store';
import {reducers} from '../../store/reducers';
import {State} from '../../store/reducers/profile';
import {EffectsModule} from '@ngrx/effects';
import {effects} from '../../store/effects';

import {UserProfilePageComponent} from './user-profile-page.component';
import {AppTranslate} from '../../app.translate';
import {MatSnackBarModule} from '@angular/material';

describe('UserProfilePageComponent', () => {
  let component: UserProfilePageComponent;
  let fixture: ComponentFixture<UserProfilePageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        CommonModule,
        HttpClientModule,
        TranslateModule,
        MatSnackBarModule,

        StoreModule.forRoot(reducers),
        EffectsModule.forRoot(effects),

        AppTranslate,
      ],
      declarations: [
        UserProfilePageComponent,
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UserProfilePageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.tabs = [];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.tabs = ['TAB1', 'TAB2', 'TAB3', 'TAB4', 'TAB5'];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.profile$ = of({loaded: true} as State);
    component.tabIndex = 0;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.profile$ = of({loaded: true} as State);
    component.tabIndex = 1;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.profile$ = of({loaded: true} as State);
    component.tabIndex = 2;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.profile$ = of({loaded: true} as State);
    component.tabIndex = 3;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
