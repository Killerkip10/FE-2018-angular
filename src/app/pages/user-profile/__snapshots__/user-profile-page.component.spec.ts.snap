// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfilePageComponent snapshot 1`] = `
<app-user-info-page
  profile$={[Function Store]}
  snackBar={[Function MatSnackBar]}
  store={[Function Store]}
  tabIndex="0"
  tabs={[Function Array]}
  users$={[Function Store]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    
    <div
      class="spinner"
    >
      <mat-spinner />
    </div>
    
  </section>
</app-user-info-page>
`;

exports[`UserProfilePageComponent snapshot 2`] = `
<app-user-info-page
  profile$={[Function Store]}
  snackBar={[Function MatSnackBar]}
  store={[Function Store]}
  tabIndex="0"
  tabs={[Function Array]}
  users$={[Function Store]}
>
  <section>
    <mat-tab-group>
      
    </mat-tab-group>
    
    <div
      class="spinner"
    >
      <mat-spinner />
    </div>
    
  </section>
</app-user-info-page>
`;

exports[`UserProfilePageComponent snapshot 3`] = `
<app-user-info-page
  profile$={[Function Store]}
  snackBar={[Function MatSnackBar]}
  store={[Function Store]}
  tabIndex="0"
  tabs={[Function Array]}
  users$={[Function Store]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
      <mat-tab />
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    
    <div
      class="spinner"
    >
      <mat-spinner />
    </div>
    
  </section>
</app-user-info-page>
`;

exports[`UserProfilePageComponent snapshot 4`] = `
<app-user-info-page
  profile$={[Function Observable]}
  snackBar={[Function MatSnackBar]}
  store={[Function Store]}
  tabIndex="0"
  tabs={[Function Array]}
  users$={[Function Store]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    
    <div
      class="container"
      ng-reflect-ng-switch="0"
    >
      
      <app-user-info />
      
      
      
    </div>
    
  </section>
</app-user-info-page>
`;

exports[`UserProfilePageComponent snapshot 5`] = `
<app-user-info-page
  profile$={[Function Observable]}
  snackBar={[Function MatSnackBar]}
  store={[Function Store]}
  tabIndex={[Function Number]}
  tabs={[Function Array]}
  users$={[Function Store]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    
    <div
      class="container"
      ng-reflect-ng-switch="1"
    >
      
      
      <app-user-form />
      
      
    </div>
    
  </section>
</app-user-info-page>
`;

exports[`UserProfilePageComponent snapshot 6`] = `
<app-user-info-page
  profile$={[Function Observable]}
  snackBar={[Function MatSnackBar]}
  store={[Function Store]}
  tabIndex={[Function Number]}
  tabs={[Function Array]}
  users$={[Function Store]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    
    <div
      class="container"
      ng-reflect-ng-switch="2"
    >
      
      
      
      <app-user-list />
      
      
      
    </div>
    
  </section>
</app-user-info-page>
`;

exports[`UserProfilePageComponent snapshot 7`] = `
<app-user-info-page
  profile$={[Function Observable]}
  snackBar={[Function MatSnackBar]}
  store={[Function Store]}
  tabIndex={[Function Number]}
  tabs={[Function Array]}
  users$={[Function Store]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    
    <div
      class="container"
      ng-reflect-ng-switch="3"
    >
      
      
      
      
      <app-user-form />
    </div>
    
  </section>
</app-user-info-page>
`;
